Source/WebKit/UIProcess/Extensions/WebExtension.cpp

```diff
@@ -240,6 +260,231 @@ void WebExtension::populatePagePropertiesIfNeeded()
+        else if (runsAtString == contentScriptsDocumentStartManifestKey)
+            injectionTime = InjectionTime::DocumentStart;
+        else if (runsAtString == contentScriptsDocumentEndManifestKey)
+            injectionTime = InjectionTime::DocumentEnd;
+        else
+            recordError(createError(Error::InvalidContentScripts, WEB_UI_STRING("Manifest `content_scripts` entry has unknown `run_at` value.", "WKWebExtensionErrorInvalidContentScripts description for unknown 'run_at' value")));
+
+        auto contentWorldType = WebExtensionContentWorldType::ContentScript;
+        auto worldString = injectedContentObject->getString(contentScriptsWorldManifestKey);
+        if (!worldString || worldString == contentScriptsIsolatedManifestKey)
+            contentWorldType = WebExtensionContentWorldType::ContentScript;
+        else if (worldString == contentScriptsMainManifestKey)
+            contentWorldType = WebExtensionContentWorldType::Main;
+        else
+            recordError(createError(Error::InvalidContentScripts, WEB_UI_STRING("Manifest `content_scripts` entry has unknown `world` value.", "WKWebExtensionErrorInvalidContentScripts description for unknown 'world' value")));
+
+        auto styleLevel = WebCore::UserStyleLevel::Author;
+        auto cssOriginString = injectedContentObject->getString(contentScriptsCSSOriginManifestKey);
+        if (!cssOriginString || cssOriginString == contentScriptsAuthorManifestKey)
+            styleLevel = WebCore::UserStyleLevel::Author;
+        else if (cssOriginString == contentScriptsUserManifestKey)
+            styleLevel = WebCore::UserStyleLevel::User;
+        else
+            recordError(createError(Error::InvalidContentScripts, WEB_UI_STRING("Manifest `content_scripts` entry has unknown `css_origin` value.", "WKWebExtensionErrorInvalidContentScripts description for unknown 'css_origin' value")));
+
+        InjectedContentData injectedContentData;
+        injectedContentData.includeMatchPatterns = WTFMove(includeMatchPatterns);
+        injectedContentData.excludeMatchPatterns = WTFMove(excludeMatchPatterns);
+        injectedContentData.injectionTime = injectionTime;
+        injectedContentData.matchesAboutBlank = matchesAboutBlank;
+        injectedContentData.injectsIntoAllFrames = injectsIntoAllFrames;
+        injectedContentData.contentWorldType = contentWorldType;
+        injectedContentData.styleLevel = styleLevel;
+        injectedContentData.scriptPaths = makeStringVector(*scriptPaths);
+        injectedContentData.styleSheetPaths = makeStringVector(*styleSheetPaths);
+        injectedContentData.includeGlobPatternStrings = makeStringVector(*includeGlobPatternStrings);
+        injectedContentData.excludeGlobPatternStrings = makeStringVector(*excludeGlobPatternStrings);
+
+        m_staticInjectedContents.append(WTFMove(injectedContentData));
+    };
+
+    for (auto injectedContentValue : *contentScriptsManifestArray) {
```

Comment:
```suggestion
    for (Ref injectedContentValue : *contentScriptsManifestArray) {
```
I think it is `Ref`.

Reply:
`error: no viable constructor or deduction guide for deduction of template arguments of 'Ref'`